frontend/src/api/prometheus/serving.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ContextResourceData, PrometheusQueryRangeResultValue } from '../../types';
+import { useContextResourceData } from '../../utilities/useContextResourceData';
+import { ModelServingMetricType } from '../../pages/modelServing/screens/metrics/ModelServingMetricsContext';
+import usePrometheusQueryRange from './usePrometheusQueryRange';
+import { TimeframeTitle } from '../../pages/modelServing/screens/types';
+import { TimeframeStep, TimeframeTime } from '../../pages/modelServing/screens/const';
+
+export const useModelServingMetrics = (
+  queries: Record<ModelServingMetricType, string>,
+  timeframe: TimeframeTitle,
+  lastUpdateTime: number,
+  setLastUpdateTime: (time: number) => void,
+): {
+  data: Record<ModelServingMetricType, ContextResourceData<PrometheusQueryRangeResultValue>>;
+  refresh: () => void;
+} => {
+  const now = Date.now();
+  let end: number;
+  // every 5 minutes refetch automatically
```

Comment:
I don't want to keep fetching the metrics data every 30 seconds so I set a `lastUpdateTime` in the context and compare it with the current time when using the hook. It works, but I am wondering if there is any better way to do this.

Reply:
I think this is similar as what @andrewballantyne did for the logout page, I think is fine the way it is now.